lib/generators/draft/account/account_generator.rb

```diff
@@ -0,0 +1,240 @@
+        filename = view_filename_with_extensions(view)
+        template filename, File.join("app/views/#{plural_table_name.underscore}", File.basename(filename))
+      end
+    end
+    
+    private
+
+    def authentication_routes
+      log :route, "Authentication routes"
+
+      route <<-RUBY.gsub(/^      /, "")
+
+        # Routes for signing up
+
+        match("/#{singular_table_name.underscore}_sign_up", { :controller => "#{singular_table_name.underscore}_sessions", :action => "registration_form", :via => "get"})
+        
+        # Routes for signing in
+        match("/#{singular_table_name.underscore}_sign_in", { :controller => "#{singular_table_name.underscore}_sessions", :action => "session_form", :via => "get"})
+        
+        match("/#{singular_table_name.underscore}_verify_credentials", { :controller => "#{singular_table_name.underscore}_sessions", :action => "add_cookie", :via => "post"})
+        
+        # Route for signing out
+        
+        match("/#{singular_table_name.underscore}_sign_out", { :controller => "#{singular_table_name.underscore}_sessions", :action => "remove_cookies", :via => "get"})
+        
+        # Route for creating account into database 
+
+        match("/insert_#{singular_table_name.underscore}", { :controller => "#{plural_table_name.underscore}", :action => "create", :via => "post" })
+        
+        # Route for editing account
+        
+        match("/edit_#{singular_table_name.underscore}", { :controller => "#{plural_table_name.underscore}", :action => "edit_account", :via => "get"})
+        
+        match("/update_#{singular_table_name.underscore}", { :controller => "#{plural_table_name.underscore}", :action => "update_account", :via => "post"})
+        
+        # Route for removing an account
+        
+        match("/cancel_#{singular_table_name.underscore}_account", { :controller => "#{plural_table_name.underscore}", :action => "delete_account", :via => "get"})
+
+
+        #------------------------------
+      RUBY
+    end
+
+    def authentication_helpers
+      log :controller, "Authentication before_actions"
+
+      application_controller <<-RUBY.gsub(/^      /, "")
+
+        before_action(:load_current_#{singular_table_name.underscore})
+        before_action(:force_#{singular_table_name.underscore}_sign_in)
+        
+        def load_current_#{singular_table_name.underscore}
+          @current_#{singular_table_name.underscore} = #{class_name.singularize}.where({ :id => session[:#{singular_table_name.underscore}_id] }).at(0)
+        end
+        
+        def force_#{singular_table_name.underscore}_sign_in
+          if @current_#{singular_table_name.underscore} == nil
+            redirect_to("/#{singular_table_name.underscore}_sign_in", { :notice => "You have to sign in first." })
+          end
+        end
+      RUBY
+    end
+
+    def basic_navbar
+      render_notifications <<-ERB.gsub(/^      /, "")

```

Comment:
If someone has already run `draft:layout` and therefore has already generated a navbar, will this add a second navbar?

Reply:
@LoganDSPrice Ah, so this actually doesn't run anymore. It's just a private method that isn't called anywhere. I will remove it.